keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -506,6 +509,30 @@ With recent conversations about kube-apiserver and enabled featureset version,
 it is possible to incorporate other version relating to the cluster, such as
 minimum kubelet version, maximum kubelet version, and enabled featureset version.
 
+#### Credentials
+
+An array of credentials to access the cluster. Each credential should be assigned 
+to a different consumer. Each credential has: 
+- a **consumer** field to indicate which consumer it is assigned to.
+- a **context** field to indicate the context name in the access information that should be used to access the cluster.
+- an **accessRef** field to point to the access information location. 
+
+The access information itself is stored in the same Kubernetes cluster on which
+the ClusterProfile API resides. However, the access information for different 
+consumers is stored in different namespaces to ensure that it is only accessible
+to the consumer for which it is intended. We recommend that the access information
+be stored in a secret object to add further protection to the access information.
+
+##### Access information format
+The access information must contain the following fields
+- **Config**: This field contains cluster access information compatible with the 
+  [kubeconfig format](https://github.com/kubernetes/kubernetes/blob/dbc2b0a5c7acc349ea71a14e49913661eaf708d2/staging/src/k8s.io/client-go/tools/clientcmd/api/types.go#L31).
```

Comment:
Drive by comment.

The recommended (and required approach for most cloud provider clusters) is to use [client-go credential plugins](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) which exist outside of the kubeconfig and can reference arbitrary config on the file system.  This config seems insufficient to express that.

Reply:
Given that plugins can do pretty much anything, I don’t think there _is_ a way to represent them and whatever they need, so I’m not sure it’s worth trying to do anything about them. If our contract is that we store `kubeconfig` files then at least that doesn’t _prevent_ the use of credential plugins, and that might well be enough.